dot/network/block_announce.go

```diff
@@ -40,7 +39,7 @@ type BlockAnnounceMessage struct {
 	Number         *big.Int
 	StateRoot      common.Hash
 	ExtrinsicsRoot common.Hash
-	Digest         types.Digest
+	Digest         scale.VaryingDataTypeSlice
```

Comment:
curious why this is a SCALE type now instead of `types.Digest`?

Reply:
This is because scale currently doesnt support custom VDTs, so in order to decode as a VDT it has to be of that type. If we add that functionality to scale, then we would be able to define a type Digest that can be treated as a VDT, but that isn't supported currently.

Reply:
As to why I used a VDT instead of the old Digest type, its because in the new scale package the buffer used in decoding/encoding is not public, which makes incremental mashalling/unmarshalling (which was pretty common in the code before) harder to do. With a Vdt representing the digest I can just call unmarshal on the header and it takes care of it, whereas if I used the old type I would have to do it in stages (that new scale doesn't really allow) using custom encode/decode logic.

So before the encoding/decoding was done in stages using custom logic as opposed to letting scale handle it all. Now it is all handled by scale without the need for custom serialization logic

Reply:
cool, this makes sense. I like it!!